### PR TITLE
Topic/xlsx files extension

### PR DESCRIPTION
**Pull request by darekbienkowski**
Description 
-----------------------------------------------------

**A set of changes to use .xlsx files (MS Office 2007+) format in a system.**

The goal is to cover all ( with one exception ) downloads and uploads of old .xls files with .xlsx files.

- For upload, where is a doable system accepts both files extension - .xls and .xlsx
- For downloads - where the previously expected download was a .xls file, now it should be .xlsx
- One exception is: Genotyping Trial Layout Intertek XLS - where I don't know how critical is .xls format for other parts of the system / pheno apps
	
All tests:  unit_fixture and unit_mech as well selenium test changed and prepared for xlsx functionality.

The server must be updated with proper libs for xlsx files. 

```
Excel::Writer::XLSX
Spreadsheet::ParseXLSX
```
Both of libraries are now in the new Debian Bullseye libs repo

https://github.com/solgenomics/perl-local-lib/tree/topic/debian_bullseye

Documentation updated.

Please inform me about any problems with uploads/downloads xlxs or xls files.

Checklist 
---------------------------------------------------------------------------------------
- [ ] Refactoring only
- [ ] Documentation only
- [ ] Fixture update only
- [ ] Bug fix
  - [ ] The relevant issue has been closed.
  - [ ] Further work is required.
- [x] New feature
  - [x] Relevant tests have been created and run.
  - [[ Data was added to the fixture
    - [ ] Data was added via a patch in `/t/data/fixture/patches/`.
  - [ ] User-Facing Change
    - [x] The user manual in `/docs` has been updated.
  - [ ] Any new Perl has been documented using **perldoc**.
  - [ ] Any new JavaScript has been documented using **JSDoc**.
  - [ ] Any new _legacy_ JavaScript has been moved from `/js` to `/js/source/legacy`.

### CODE REVIEW

File: lib/SGN/Controller/Search/Cross.pm [hunk 1]
```diff
@@ -76,12 +76,12 @@ sub download_cross_entries : Path('/search/download_cross_entries') Args(0) {
 
     my $cross_properties_string = $c->config->{cross_properties};
     my @cross_properties = split ',', $cross_properties_string;
-    my $file_format = "xls";
+    my $file_format = "xlsx";
 
     my $time = DateTime->now();
     my $timestamp = $time->ymd();
     my $dir = $c->tempfiles_subdir('download');
-    my $temp_file_name = "cross_entries". "XXXX";
```
Comment (nickmorales) [inline lib/SGN/Controller/Search/Cross.pm:84]: it is probably better to keep the XXXX for random file generation

File: lib/CXGN/Trial/ParseUpload/Plugin/TrialExcelFormat.pm [hunk 2]
```diff
@@ -368,16 +371,6 @@ sub _validate_with_plugin {
         push @error_messages, "Cell A".$seen_plot_names{$r->uniquename}.": plot name already exists: ".$r->uniquename;
     }
 
-    # check for multiple plots at the same position
-    foreach my $key (keys %seen_plot_keys) {
-        my $plots = $seen_plot_keys{$key};
-        my $count = scalar(@{$plots});
-        if ( $count > 1 ) {
-            my @pos = split('-', $key);
-            push @error_messages, "More than 1 plot is assigned to the position row=" . $pos[0] . " col=" . $pos[1] . " plots=" . join(',', @$plots);
```
Comment (nickmorales) [inline lib/CXGN/Trial/ParseUpload/Plugin/TrialExcelFormat.pm:377]: Should this be removed? Just checking

File: lib/CXGN/Trial/ParseUpload/Plugin/MultipleTrialDesignExcelFormat.pm [hunk 3]
```diff
@@ -587,18 +584,6 @@ sub _validate_with_plugin {
...
   }
 
-  ## PLOT POSITION OVERALL VALIDATION
-  foreach my $tk (keys %seen_plot_keys) {
-      foreach my $pk (keys %{$seen_plot_keys{$tk}} ) {
-          my $plots = $seen_plot_keys{$tk}{$pk};
-          my $count = scalar(@{$plots});
-          if ( $count > 1 ) {
-              my @pos = split('-', $pk);
-              push @error_messages, "More than 1 plot is assigned to the position row=" . $pos[0] . " col=" . $pos[1] . " trial=" . $tk . " plots=" . join(',', @$plots);
```
Comment (nickmorales) [inline lib/CXGN/Trial/ParseUpload/Plugin/MultipleTrialDesignExcelFormat.pm:597]: Should this be removed? Just checking

**Review by nickmorales — COMMENTED** [review]
:+1:

File: lib/SGN/Controller/Search/Cross.pm (hunk 1 above)
Comment (darekbienkowski) [reply lib/SGN/Controller/Search/Cross.pm:84]: Great catch - I will check it. Thank you.

File: lib/CXGN/Trial/ParseUpload/Plugin/TrialExcelFormat.pm (hunk 2 above)
Comment (darekbienkowski) [reply lib/CXGN/Trial/ParseUpload/Plugin/TrialExcelFormat.pm:377]: That one is awkward - it is not connected with any .xlsx functionality. Looks more like some problem with rebasing / branches. I will check why it is even there.

File: lib/CXGN/Trial/ParseUpload/Plugin/MultipleTrialDesignExcelFormat.pm (hunk 3 above)
Comment (darekbienkowski) [reply lib/CXGN/Trial/ParseUpload/Plugin/MultipleTrialDesignExcelFormat.pm:597]: That one is awkward - it is not connected with any .xlsx functionality. Looks more like some problem with rebasing / branches. I will check why it is even there.

**Pull request closed without merging**